gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -25,39 +24,38 @@
 final class CloudStorageReadChannel implements SeekableByteChannel {
 
   static CloudStorageReadChannel create(
-      GcsService gcsService, GcsFilename file, long position) throws IOException {
+      Storage gcsStorage, BlobId file, long position) throws IOException {
     // XXX: Reading size and opening file should be atomic.
-    long size = fetchSize(gcsService, file);
-    return new CloudStorageReadChannel(gcsService, file, position, size,
-        gcsService.openReadChannel(file, position));
+    long size = fetchSize(gcsStorage, file);
+    ReadChannel channel = gcsStorage.reader(file);
+    if (position > 0) {
+      channel.seek((int) position);
+    }
+    return new CloudStorageReadChannel(position, size, channel);
   }
 
-  private final GcsService gcsService;
-  private final GcsFilename file;
+  private final ReadChannel gcsChannel;
```

Comment:
Could you change the name of this field back to `channel`? It's creating a bunch of diff noise.